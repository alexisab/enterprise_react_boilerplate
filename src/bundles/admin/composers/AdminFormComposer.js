import { pure, compose, withState, withHandlers } from 'recompose';
import { graphql } from 'react-apollo';
import { addAdminMutation, adminListQuery } from '../graphql';
import { formatErrors } from '../../../helpers/form';
import { Notification } from '../../../shared';

const AdminFormComposer = compose(
  graphql(addAdminMutation),
  withState('formData', 'setFormData', { id: '', name: '', email: '' }),
  withState('formErrors', 'setErrors', { id: '', name: '', email: '' }),
  withHandlers({
    handleSave: props => () => {
      const { formData, setErrors, mutate, Dialog } = props;
      mutate({
        variables: formData,
        update: (store, { data: { addAdmin } }) => {
          if (addAdmin.errors.length <= 0) {
            const data = store.readQuery({ query: adminListQuery });
            data.admins.push(addAdmin);
            store.writeQuery({ query: adminListQuery, data });
          } else {
            setErrors(Object.assign({ name: '' }, formatErrors(addAdmin.errors)));
          }
        }
      })
        .then(({ data: { addAdmin } }) => {
          if (addAdmin.errors.length === 0) {
            Notification('success', 'Admin Invited', `We have invited ${addAdmin.name} to be an admin`);
            Dialog.close();
          }
        });
    },
    handleClose: props => () => {
      props.Dialog.close();
    }
  }),
  pure
)

export default AdminFormComposer;
