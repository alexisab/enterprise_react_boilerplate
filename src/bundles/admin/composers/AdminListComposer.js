import { pure, compose, withHandlers } from 'recompose';
import { graphql } from 'react-apollo';
import { adminListQuery, sendInviteMutation } from '../graphql';
import { displayLoadingState } from '../../../helpers/compose';
import { Notification } from '../../../shared';

const AdminListComposer = compose(
  graphql(adminListQuery),
  displayLoadingState,
  graphql(sendInviteMutation, { name: 'sendInviteMutate'} ),
  withHandlers({
    handleDelete: props => (id) => {
      console.log('handleDelete');
    },
    sendInvite: props => (id) => {
      const { sendInviteMutate } = props;
      sendInviteMutate({
        variables: { id }
      }).then(({ data: { sendInvite } }) => {
        if (sendInvite.errors.length <= 0) {
          Notification('success', 'Admin Invited', `We have send another invitation email to ${sendInvite.name}.`);
        }
      })
      console.log('send Invite')
    }
  }),
  pure
);

export default AdminListComposer;
