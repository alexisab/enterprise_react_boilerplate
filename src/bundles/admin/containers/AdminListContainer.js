import React from 'react';
import { Button } from 'antd';
import AdminForm from '../components/AdminForm';

const AdminListContainer = (props) => {
  const { Dialog } = props;

  return (
    <div>
      <h1>All Admins</h1>
      <Button
        type="primary"
        onClick={(e) => {
          e.preventDefault;
          Dialog.open(
            <AdminForm Dialog={Dialog} />,
            { title: 'Add New Admin' }
          );
        }}>Add Admin</Button>
    </div>
  );
};

export default AdminListContainer;
