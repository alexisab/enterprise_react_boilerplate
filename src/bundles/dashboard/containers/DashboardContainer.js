import React from "react";
import PropTypes from "prop-types";
import Dashboard from "../components/Dashboard";

const DashboardContainer = ({ currentUser }) => (
  <div>
    <Dashboard />
  </div>
);

DashboardContainer.propTypes = {
  currentUser: PropTypes.shape({
    name: PropTypes.string,
    admin: PropTypes.bool,
    super_admin: PropTypes.bool,
    id: PropTypes.string
  }).isRequired
};

export default DashboardContainer;
