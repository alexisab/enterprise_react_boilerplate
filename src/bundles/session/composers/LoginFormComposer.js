import { pure, compose, withHandlers, withState } from "recompose";
import { graphql } from "react-apollo";
import LoginForm from "../components/LoginForm";
import { displayLoadingState } from "../../../helpers/compose";
import { authenticateUser } from "../graphql";
import { signIn, signOut } from "../../../helpers/auth";

const LoginFormComposer = compose(
  graphql(authenticateUser),
  withState("formData", "setFormData", { email: "", password: "" }),
  withHandlers({
    handleSave: props => () => {
      const { formData, mutate, history } = props;
      mutate({
        variables: { ...formData }
      }).then(({ data }) => {
        if (data.authenticate) {
          signIn(data.authenticate);
          history.push("/dashboard");
        } else {
          signOut();
        }
      });
    }
  }),
  displayLoadingState,
  pure
)(LoginForm);

export default LoginFormComposer;
