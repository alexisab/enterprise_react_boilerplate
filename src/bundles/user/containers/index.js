import SignUpContainer from './SignUpContainer';
import AcceptInvitationContainer from './AcceptInvitationContainer';
import ForgetPasswordContainer from './ForgetPasswordContainer';
import ResetPasswordContainer from './ResetPasswordContainer';

export { SignUpContainer, AcceptInvitationContainer, ForgetPasswordContainer, ResetPasswordContainer };
