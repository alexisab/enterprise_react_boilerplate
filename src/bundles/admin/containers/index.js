import AdminListContainer from './AdminListContainer';

export { AdminListContainer }
