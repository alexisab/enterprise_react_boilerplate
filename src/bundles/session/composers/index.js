import LoginFormComposer from './LoginFormComposer';
export { LoginFormComposer }
