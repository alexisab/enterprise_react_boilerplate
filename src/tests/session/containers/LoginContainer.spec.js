import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { LoginContainer } from "../../../bundles/session/containers";
import { LoginFormComposer as LoginForm } from "../../../bundles/session/composers";

const history = {
  go: () => {},
  push: () => {}
};

test("it should match snapshot", () => {
  const component = renderer.create(
    <MemoryRouter>
      <LoginContainer history={history} />
    </MemoryRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("it should have the h1 element with the name 'Log in'", () => {
  const wrapper = shallow(<LoginContainer history={history} />);
  expect(wrapper.find("h1").text()).toEqual("Log in");
});

test("it should contain the LoginFormComposer as LoginForm", () => {
  const wrapper = shallow(<LoginContainer history={history} />);
  expect(wrapper.find(LoginForm).length).toEqual(1);
});
