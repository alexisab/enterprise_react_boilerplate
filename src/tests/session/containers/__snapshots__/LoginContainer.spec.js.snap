// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<div
  className="login-page"
>
  <h1
    className="title"
  >
    Log in
  </h1>
  <div
    className="login-form"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <span
                className="ant-input-affix-wrapper"
                style={undefined}
              >
                <span
                  className="ant-input-prefix"
                >
                  <i
                    className="anticon anticon-mail"
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                  />
                </span>
                <input
                  className="ant-input"
                  disabled={false}
                  id="email-field"
                  name="Email"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Email"
                  style={null}
                  type="text"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <span
                className="ant-input-affix-wrapper"
                style={undefined}
              >
                <span
                  className="ant-input-prefix"
                >
                  <i
                    className="anticon anticon-lock"
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                  />
                </span>
                <input
                  className="ant-input"
                  disabled={false}
                  name="Password"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Password"
                  style={null}
                  type="password"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <button
                className="ant-btn login-form-button ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Log in
                </span>
              </button>
              <div
                className="form-footer"
              >
                <a
                  href="/forget_password"
                  onClick={[Function]}
                >
                  Forgot password?
                </a>
                <br />
                <a
                  href="/signup"
                >
                  Do not have account? Sign up here
                </a>
              </div>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
